File: lib_layerd_xpl_codedom_cs/CDOM_XplNewExpression.cs
/*-------------------------------------------------
 *
 *	Este archivo fue generado automáticamente.
 *	Fecha: 8/11/2011 4:27:20 PM
 *
 *	Generado por Zoe CodeDOM Generator para C#.
 *	COPYRIGHT 2002,2005-2006. por Alexis Ferreyra.
 *
 *------------------------------------------------*/

using System;
using System.Xml;
using System.IO;

namespace LayerD.CodeDOM{

[Serializable]
public class XplNewExpression:  XplNode{

	#region Variables privadas para atributos y elementos
	string p_GCName;
	string p_doc;
	string p_helpURL;
	string p_ldsrc;
	bool p_iny;
	string p_inydata;
	string p_inyby;
	string p_lddata;
	//Variables para Elementos de Secuencia
	XplType p_type;
	XplInitializerList p_init;
	XplExpressionlist p_GCParams;
	#endregion

	#region Region de Constructores Publicos
	public XplNewExpression(){
		p_GCName = "default";
		p_doc = "";
		p_helpURL = "";
		p_ldsrc = "";
		p_iny = false;
		p_inydata = "";
		p_inyby = "";
		p_lddata = "";
		p_type=null;
		p_init=null;
		p_GCParams=null;
}
	public XplNewExpression(string n_GCName){
		p_GCName = "default";
		p_doc = "";
		p_helpURL = "";
		p_ldsrc = "";
		p_iny = false;
		p_inydata = "";
		p_inyby = "";
		p_lddata = "";
		set_GCName(n_GCName);
		p_type=null;
		p_init=null;
		p_GCParams=null;
	}
	public XplNewExpression(XplType n_type){
		p_GCName = "default";
		p_doc = "";
		p_helpURL = "";
		p_ldsrc = "";
		p_iny = false;
		p_inydata = "";
		p_inyby = "";
		p_lddata = "";
		p_type=null;
		p_init=null;
		p_GCParams=null;
		set_type(n_type);
	}
	public XplNewExpression(string n_GCName, XplType n_type){
		p_GCName = "default";
		p_doc = "";
		p_helpURL = "";
		p_ldsrc = "";
		p_iny = false;
		p_inydata = "";
		p_inyby = "";
		p_lddata = "";
		set_GCName(n_GCName);
		p_type=null;
		p_init=null;
		p_GCParams=null;
		set_type(n_type);
	}
	public XplNewExpression(string n_GCName, string n_doc, string n_helpURL, string n_ldsrc, bool n_iny, string n_inydata, string n_inyby, string n_lddata){
		set_GCName(n_GCName);
		set_doc(n_doc);
		set_helpURL(n_helpURL);
		set_ldsrc(n_ldsrc);
		set_iny(n_iny);
		set_inydata(n_inydata);
		set_inyby(n_inyby);
		set_lddata(n_lddata);
		p_type=null;
		p_init=null;
		p_GCParams=null;
	}
	public XplNewExpression(XplType n_type, XplInitializerList n_init, XplExpressionlist n_GCParams){
		p_GCName = "default";
		p_doc = "";
		p_helpURL = "";
		p_ldsrc = "";
		p_iny = false;
		p_inydata = "";
		p_inyby = "";
		p_lddata = "";
		p_type=null;
		p_init=null;
		p_GCParams=null;
		set_type(n_type);
		set_init(n_init);
		set_GCParams(n_GCParams);
	}
	public XplNewExpression(string n_GCName, string n_doc, string n_helpURL, string n_ldsrc, bool n_iny, string n_inydata, string n_inyby, string n_lddata, XplType n_type){
		set_GCName(n_GCName);
		set_doc(n_doc);
		set_helpURL(n_helpURL);
		set_ldsrc(n_ldsrc);
		set_iny(n_iny);
		set_inydata(n_inydata);
		set_inyby(n_inyby);
		set_lddata(n_lddata);
		set_GCName(n_GCName);
		p_type=null;
		p_init=null;
		p_GCParams=null;
	}
	public XplNewExpression(string n_GCName, XplType n_type, XplInitializerList n_init, XplExpressionlist n_GCParams){
		p_GCName = "default";
		p_doc = "";
		p_helpURL = "";
		p_ldsrc = "";
		p_iny = false;
		p_inydata = "";
		p_inyby = "";
		p_lddata = "";
		set_GCName(n_GCName);
		p_type=null;
		p_init=null;
		p_GCParams=null;
		set_type(n_type);
		set_init(n_init);
		set_GCParams(n_GCParams);
	}
	public XplNewExpression(string n_GCName, string n_doc, string n_helpURL, string n_ldsrc, bool n_iny, string n_inydata, string n_inyby, string n_lddata, XplType n_type, XplInitializerList n_init, XplExpressionlist n_GCParams){
		set_GCName(n_GCName);
		set_doc(n_doc);
		set_helpURL(n_helpURL);
		set_ldsrc(n_ldsrc);
		set_iny(n_iny);
		set_inydata(n_inydata);
		set_inyby(n_inyby);
		set_lddata(n_lddata);
		p_type=null;
		p_init=null;
		p_GCParams=null;
		set_type(n_type);
		set_init(n_init);
		set_GCParams(n_GCParams);
	}
	#endregion

	#region Destructor
/*	public ~XplNewExpression(){
		Se supone que .NET no necesita destructores sólo para liberar memoria	}*/
	#endregion

	#region Funciones Sobreescritas de XplNode
	public override XplNode Clone(){
		XplNewExpression copy = new XplNewExpression();
		copy.set_GCName(this.p_GCName);
		copy.set_doc(this.p_doc);
		copy.set_helpURL(this.p_helpURL);
		copy.set_ldsrc(this.p_ldsrc);
		copy.set_iny(this.p_iny);
		copy.set_inydata(this.p_inydata);
		copy.set_inyby(this.p_inyby);
		copy.set_lddata(this.p_lddata);
		if(p_type!=null)copy.set_type((XplType)p_type.Clone());
		if(p_init!=null)copy.set_init((XplInitializerList)p_init.Clone());
		if(p_GCParams!=null)copy.set_GCParams((XplExpressionlist)p_GCParams.Clone());
		copy.set_ElementName(this.get_ElementName());
		return (XplNode)copy;
	}

	public override CodeDOMTypes get_TypeName(){return CodeDOMTypes.XplNewExpression;}

	public override bool Write(XplWriter writer){
		bool result=true;
		//Escribo el encabezado del elemento
		writer.WriteStartElement( this.get_ElementName() );
		//Escribo los atributos del elemento
		if(p_GCName != "default")
			writer.WriteAttributeString( "GCName" ,ZoeHelper .Att_ToString(p_GCName) );
		if(p_doc != "")
			writer.WriteAttributeString( "doc" ,ZoeHelper .Att_ToString(p_doc) );
		if(p_helpURL != "")
			writer.WriteAttributeString( "helpURL" ,ZoeHelper .Att_ToString(p_helpURL) );
		if(p_ldsrc != "")
			writer.WriteAttributeString( "ldsrc" ,ZoeHelper .Att_ToString(p_ldsrc) );
		if(p_iny != false)
			writer.WriteAttributeString( "iny" ,ZoeHelper .Att_ToString(p_iny) );
		if(p_inydata != "")
			writer.WriteAttributeString( "inydata" ,ZoeHelper .Att_ToString(p_inydata) );
		if(p_inyby != "")
			writer.WriteAttributeString( "inyby" ,ZoeHelper .Att_ToString(p_inyby) );
		if(p_lddata != "")
			writer.WriteAttributeString( "lddata" ,ZoeHelper .Att_ToString(p_lddata) );
		//Escribo recursivamente cada elemento hijo
		if(p_type!=null)if(!p_type.Write(writer))result=false;
		if(p_init!=null)if(!p_init.Write(writer))result=false;
		if(p_GCParams!=null)if(!p_GCParams.Write(writer))result=false;
		//Cierro el elemento
		writer.WriteEndElement();
		return result;
	}

	public override XplNode Read(XplReader reader){
		this.set_ElementName( reader.Name );
		//Lectura de Atributos
		if( reader.HasAttributes ){
			for(int i=0; i < reader.AttributeCount ; i++){
				reader.MoveToAttribute(i);
				switch(reader.Name){
					case "GCName":
						this.set_GCName(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "doc":
						this.set_doc(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "helpURL":
						this.set_helpURL(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "ldsrc":
						this.set_ldsrc(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "iny":
						this.set_iny(ZoeHelper .StringAtt_To_BOOL(reader.Value));
						break;
					case "inydata":
						this.set_inydata(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "inyby":
						this.set_inyby(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					case "lddata":
						this.set_lddata(ZoeHelper .StringAtt_To_STRING(reader.Value));
						break;
					default:
						throw new CodeDOM_Exception("Linea: "+reader.LineNumber+".Atributo '"+reader.Name+"' invalido en elemento '"+this.get_ElementName()+"'.");
				}
			}
			reader.MoveToElement();
		}
		//Lectura de Elementos
		//Borro todo posible nodo en memoria
		this.p_type=null;
		this.p_init=null;
		this.p_GCParams=null;
		if(!reader.IsEmptyElement){
		reader.Read();
		while(reader.NodeType!=XmlNodeType.EndElement){
			XplNode tempNode=null;
			switch(reader.NodeType){
				case XmlNodeType.Element:
					switch(reader.Name){
					case "type":
						tempNode = new XplType();
						tempNode.Read(reader);
						if(this.get_type()!=null)throw new CodeDOM_Exception("Linea: "+reader.LineNumber+".Nodo '"+reader.Name+"' repetido como hijo de elemento '"+this.get_ElementName()+"'.");
						this.set_type((XplType)tempNode);
						break;
					case "init":
						tempNode = new XplInitializerList();
						tempNode.Read(reader);
						if(this.get_init()!=null)throw new CodeDOM_Exception("Linea: "+reader.LineNumber+".Nodo '"+reader.Name+"' repetido como hijo de elemento '"+this.get_ElementName()+"'.");
						this.set_init((XplInitializerList)tempNode);
						break;
					case "GCParams":
						tempNode = new XplExpressionlist();
						tempNode.Read(reader);
						if(this.get_GCParams()!=null)throw new CodeDOM_Exception("Linea: "+reader.LineNumber+".Nodo '"+reader.Name+"' repetido como hijo de elemento '"+this.get_ElementName()+"'.");
						this.set_GCParams((XplExpressionlist)tempNode);
						break;
					default:
						throw new CodeDOM_Exception("Linea: "+reader.LineNumber+".Nombre de nodo '"+reader.Name+"' inesperado como hijo de elemento '"+this.get_ElementName()+"'.");
					}
					break;
				case XmlNodeType.EndElement:
					break;
				case XmlNodeType.Text:
					throw new CodeDOM_Exception("Linea: "+reader.LineNumber+".No se esperaba texto en este nodo.");
				default:
					break;
			}
			reader.Read();
		}
		}
		return this;
	}
	public override bool BinaryWrite(XplBinaryWriter writer){
		bool result=true;
		//Escribo el ID y el nombre del elemento
		writer.Write((int) 150 );
		writer.Write( this.get_ElementName() );
		//Escribo los atributos del elemento
		writer.Write( p_GCName );
		writer.Write( p_doc );
		writer.Write( p_helpURL );
		writer.Write( p_ldsrc );
		writer.Write( p_iny );
		writer.Write( p_inydata );
		writer.Write( p_inyby );
		writer.Write( p_lddata );
		//Escribo recursivamente cada elemento hijo
		if(p_type!=null){
			writer.Write((int)1);
			if(!p_type.BinaryWrite(writer))result=false;
		}
		else {
			writer.Write((int)0);
		}
		if(p_init!=null){
			writer.Write((int)1);
			if(!p_init.BinaryWrite(writer))result=false;
		}
		else {
			writer.Write((int)0);
		}
		if(p_GCParams!=null){
			writer.Write((int)1);
			if(!p_GCParams.BinaryWrite(writer))result=false;
		}
		else {
			writer.Write((int)0);
		}
		return result;
	}

	public override XplNode BinaryRead(XplBinaryReader reader){
		this.set_ElementName( reader.ReadString() );
		//Lectura de Atributos
		p_GCName = reader.ReadString();
		p_doc = reader.ReadString();
		p_helpURL = reader.ReadString();
		p_ldsrc = reader.ReadString();
		p_iny = reader.ReadBoolean();
		p_inydata = reader.ReadString();
		p_inyby = reader.ReadString();
		p_lddata = reader.ReadString();
		//Lectura de Elementos
		if(reader.ReadInt32()==1){
			p_type = (XplType)CDOM_BinaryNodeReader.Read(reader, this.get_Parent());
		}
		if(reader.ReadInt32()==1){
			p_init = (XplInitializerList)CDOM_BinaryNodeReader.Read(reader, this.get_Parent());
		}
		if(reader.ReadInt32()==1){
			p_GCParams = (XplExpressionlist)CDOM_BinaryNodeReader.Read(reader, this.get_Parent());
		}
		return this;
	}
/// <summary>
/// Adds arg2 node to the end of the Children collection of arg1 node.
/// </summary>
/// <param name="arg1">Node to which arg2 will be added as children.</param>
/// <param name="arg2">Node to add to arg1 as children.</param>
/// <returns>arg1 node or CodeDOM_Exception if arg1 doesn't has children.</returns>
public static XplNewExpression operator +(XplNewExpression arg1, XplNode arg2)
{
    if (arg1.Children() != null)
        arg1.Children().InsertAtEnd(arg2);
    else
        throw new CodeDOM_Exception("Node doesn't has children. Node can't be added.");
    return arg1;
}
/// <summary>
/// Adds arg2 nodes to the end of the Children collection of arg1 node.
/// </summary>
/// <param name="arg1">Node to which arg2 will be added as children.</param>
/// <param name="arg2">Nodes to add to arg1 as children.</param>
/// <returns>arg1 node or CodeDOM_Exception if arg1 doesn't has children.</returns>
public static XplNewExpression operator +(XplNewExpression arg1, XplNodeList arg2)
{
    if (arg1.Children() != null)
        arg1.Children().InsertAtEnd(arg2);
    else
        throw new CodeDOM_Exception("Node doesn't has children. Node can't be added.");
    return arg1;
}
	#endregion

	#region Funciones para obtener Atributos y Elementos
	public string get_GCName(){
		return p_GCName;
	}
	public override string get_doc(){
		return p_doc;
	}
	public override string get_helpURL(){
		return p_helpURL;
	}
	public override string get_ldsrc(){
		return p_ldsrc;
	}
	public override bool get_iny(){
		return p_iny;
	}
	public override string get_inydata(){
		return p_inydata;
	}
	public override string get_inyby(){
		return p_inyby;
	}
	public override string get_lddata(){
		return p_lddata;
	}
	//Funciones para obtener Elementos de Secuencia
	public XplType get_type(){
		return p_type;
	}
	public XplInitializerList get_init(){
		return p_init;
	}
	public XplExpressionlist get_GCParams(){
		return p_GCParams;
	}
	#endregion

	#region Funcion ChildNodes() y Attributes()
	public override string[] Attributes(){
		string[] ret = new string[8];
		ret[0] = "GCName";
		ret[1] = "doc";
		ret[2] = "helpURL";
		ret[3] = "ldsrc";
		ret[4] = "iny";
		ret[5] = "inydata";
		ret[6] = "inyby";
		ret[7] = "lddata";
		return ret;
	}
	public override string AttributeValue(string attributeName){
		if (attributeName=="GCName") return p_GCName.ToString();
		if (attributeName=="doc") return p_doc.ToString();
		if (attributeName=="helpURL") return p_helpURL.ToString();
		if (attributeName=="ldsrc") return p_ldsrc.ToString();
		if (attributeName=="iny") return p_iny.ToString();
		if (attributeName=="inydata") return p_inydata.ToString();
		if (attributeName=="inyby") return p_inyby.ToString();
		if (attributeName=="lddata") return p_lddata.ToString();
		return null;
	}
	public override XplNodeList ChildNodes(){
		XplNodeList list = new XplNodeList();
		list.InsertAtEnd(p_type);
		list.InsertAtEnd(p_init);
		list.InsertAtEnd(p_GCParams);
		return list;
	}
	#endregion

	#region Funciones para setear Atributos y Elementos
	public string set_GCName(string new_GCName){
		string back_GCName = p_GCName;
		p_GCName = new_GCName;
		return back_GCName;
	}
	public override string set_doc(string new_doc){
		string back_doc = p_doc;
		p_doc = new_doc;
		return back_doc;
	}
	public override string set_helpURL(string new_helpURL){
		string back_helpURL = p_helpURL;
		p_helpURL = new_helpURL;
		return back_helpURL;
	}
	public override string set_ldsrc(string new_ldsrc){
		string back_ldsrc = p_ldsrc;
		p_ldsrc = new_ldsrc;
		return back_ldsrc;
	}
	public override bool set_iny(bool new_iny){
		bool back_iny = p_iny;
		p_iny = new_iny;
		return back_iny;
	}
	public override string set_inydata(string new_inydata){
		string back_inydata = p_inydata;
		p_inydata = new_inydata;
		return back_inydata;
	}
	public override string set_inyby(string new_inyby){
		string back_inyby = p_inyby;
		p_inyby = new_inyby;
		return back_inyby;
	}
	public override string set_lddata(string new_lddata){
		string back_lddata = p_lddata;
		p_lddata = new_lddata;
		return back_lddata;
	}

	//Funciones para setear Elementos de Secuencia
	public XplType set_type(XplType new_type){
		XplType back_type = p_type;
		p_type = new_type;
		if(p_type!=null){
			p_type.set_ElementName("type");
			p_type.set_Parent(this);
		}
		return back_type;
	}
	public XplInitializerList set_init(XplInitializerList new_init){
		XplInitializerList back_init = p_init;
		p_init = new_init;
		if(p_init!=null){
			p_init.set_ElementName("init");
			p_init.set_Parent(this);
		}
		return back_init;
	}
	public XplExpressionlist set_GCParams(XplExpressionlist new_GCParams){
		XplExpressionlist back_GCParams = p_GCParams;
		p_GCParams = new_GCParams;
		if(p_GCParams!=null){
			p_GCParams.set_ElementName("GCParams");
			p_GCParams.set_Parent(this);
		}
		return back_GCParams;
	}
	#endregion

	#region Funciones para obtener nuevos nodos hijos
	static public XplType new_type(){
		XplType node = new XplType();
		node.set_ElementName("type");
		return node;
	}
	static public XplInitializerList new_init(){
		XplInitializerList node = new XplInitializerList();
		node.set_ElementName("init");
		return node;
	}
	static public XplExpressionlist new_GCParams(){
		XplExpressionlist node = new XplExpressionlist();
		node.set_ElementName("GCParams");
		return node;
	}
	#endregion

}	//Fin declaración de Clase

}

/*-------Fin de Archivo Generado------------------*/

File: trunk/CodeDOM/CDOM_XplNewExpression.cpp
/*-------------------------------------------------
 *
 *	Este archivo fue generado automáticamente.
 *	Fecha: 09/04/2009 17:21:33
 *
 *	Generado por Zoe CodeDOM Generator.
 *	COPYRIGHT 2002. por Alexis Ferreyra.
 *
 *------------------------------------------------*/

#ifndef __LAYERD_CODEDOM_ZOENEWEXPRESSION_V1_0_CPP
#define __LAYERD_CODEDOM_ZOENEWEXPRESSION_V1_0_CPP
#include "CDOM_XplNewExpression.h"

namespace CodeDOM{

//Region de Constructores Publicos
XplNewExpression::XplNewExpression(){
	p_GCName = DT("default");
	p_doc = DT_EMPTY;
	p_helpURL = DT_EMPTY;
	p_ldsrc = DT_EMPTY;
	p_iny = false;
	p_inydata = DT_EMPTY;
	p_inyby = DT_EMPTY;
	p_lddata = DT_EMPTY;
	p_type=NULL;
	p_init=NULL;
	p_GCParams=NULL;
}
XplNewExpression::XplNewExpression(string n_GCName){
	p_GCName = DT("default");
	p_doc = DT_EMPTY;
	p_helpURL = DT_EMPTY;
	p_ldsrc = DT_EMPTY;
	p_iny = false;
	p_inydata = DT_EMPTY;
	p_inyby = DT_EMPTY;
	p_lddata = DT_EMPTY;
	set_GCName(n_GCName);
	p_type=NULL;
	p_init=NULL;
	p_GCParams=NULL;
}
XplNewExpression::XplNewExpression(XplType* n_type){
	p_GCName = DT("default");
	p_doc = DT_EMPTY;
	p_helpURL = DT_EMPTY;
	p_ldsrc = DT_EMPTY;
	p_iny = false;
	p_inydata = DT_EMPTY;
	p_inyby = DT_EMPTY;
	p_lddata = DT_EMPTY;
	p_type=NULL;
	p_init=NULL;
	p_GCParams=NULL;
	set_type(n_type);
}
XplNewExpression::XplNewExpression(string n_GCName, XplType* n_type){
	p_GCName = DT("default");
	p_doc = DT_EMPTY;
	p_helpURL = DT_EMPTY;
	p_ldsrc = DT_EMPTY;
	p_iny = false;
	p_inydata = DT_EMPTY;
	p_inyby = DT_EMPTY;
	p_lddata = DT_EMPTY;
	set_GCName(n_GCName);
	p_type=NULL;
	p_init=NULL;
	p_GCParams=NULL;
	set_type(n_type);
}
XplNewExpression::XplNewExpression(string n_GCName, string n_doc, string n_helpURL, string n_ldsrc, bool n_iny, string n_inydata, string n_inyby, string n_lddata){
	set_GCName(n_GCName);
	set_doc(n_doc);
	set_helpURL(n_helpURL);
	set_ldsrc(n_ldsrc);
	set_iny(n_iny);
	set_inydata(n_inydata);
	set_inyby(n_inyby);
	set_lddata(n_lddata);
	p_type=NULL;
	p_init=NULL;
	p_GCParams=NULL;
}
XplNewExpression::XplNewExpression(XplType* n_type, XplInitializerList* n_init, XplExpressionlist* n_GCParams){
	p_GCName = DT("default");
	p_doc = DT_EMPTY;
	p_helpURL = DT_EMPTY;
	p_ldsrc = DT_EMPTY;
	p_iny = false;
	p_inydata = DT_EMPTY;
	p_inyby = DT_EMPTY;
	p_lddata = DT_EMPTY;
	p_type=NULL;
	p_init=NULL;
	p_GCParams=NULL;
	set_type(n_type);
	set_init(n_init);
	set_GCParams(n_GCParams);
}
XplNewExpression::XplNewExpression(string n_GCName, string n_doc, string n_helpURL, string n_ldsrc, bool n_iny, string n_inydata, string n_inyby, string n_lddata, XplType* n_type){
	set_GCName(n_GCName);
	set_doc(n_doc);
	set_helpURL(n_helpURL);
	set_ldsrc(n_ldsrc);
	set_iny(n_iny);
	set_inydata(n_inydata);
	set_inyby(n_inyby);
	set_lddata(n_lddata);
	set_GCName(n_GCName);
	p_type=NULL;
	p_init=NULL;
	p_GCParams=NULL;
}
XplNewExpression::XplNewExpression(string n_GCName, XplType* n_type, XplInitializerList* n_init, XplExpressionlist* n_GCParams){
	p_GCName = DT("default");
	p_doc = DT_EMPTY;
	p_helpURL = DT_EMPTY;
	p_ldsrc = DT_EMPTY;
	p_iny = false;
	p_inydata = DT_EMPTY;
	p_inyby = DT_EMPTY;
	p_lddata = DT_EMPTY;
	set_GCName(n_GCName);
	p_type=NULL;
	p_init=NULL;
	p_GCParams=NULL;
	set_type(n_type);
	set_init(n_init);
	set_GCParams(n_GCParams);
}
XplNewExpression::XplNewExpression(string n_GCName, string n_doc, string n_helpURL, string n_ldsrc, bool n_iny, string n_inydata, string n_inyby, string n_lddata, XplType* n_type, XplInitializerList* n_init, XplExpressionlist* n_GCParams){
	set_GCName(n_GCName);
	set_doc(n_doc);
	set_helpURL(n_helpURL);
	set_ldsrc(n_ldsrc);
	set_iny(n_iny);
	set_inydata(n_inydata);
	set_inyby(n_inyby);
	set_lddata(n_lddata);
	p_type=NULL;
	p_init=NULL;
	p_GCParams=NULL;
	set_type(n_type);
	set_init(n_init);
	set_GCParams(n_GCParams);
}
//Destructor
XplNewExpression::~XplNewExpression(){
#ifdef __LAYERD_CODEDOM_DEBUG_DESTRUCTORS
	__LAYERD_CODEDOM_DEBUG_OUTPUT("En el Destructor de 'XplNewExpression'");
#endif
	//Variables para Elementos de Secuencia
	if(p_type!=NULL)delete p_type;
	if(p_init!=NULL)delete p_init;
	if(p_GCParams!=NULL)delete p_GCParams;
}

//Funciones Sobreescritas de XplNode
XplNode* XplNewExpression::Clone(){
	XplNewExpression* copy = new XplNewExpression();
	copy->set_GCName(this->p_GCName);
	copy->set_doc(this->p_doc);
	copy->set_helpURL(this->p_helpURL);
	copy->set_ldsrc(this->p_ldsrc);
	copy->set_iny(this->p_iny);
	copy->set_inydata(this->p_inydata);
	copy->set_inyby(this->p_inyby);
	copy->set_lddata(this->p_lddata);
	if(p_type!=NULL)copy->set_type((XplType*)p_type->Clone());
	if(p_init!=NULL)copy->set_init((XplInitializerList*)p_init->Clone());
	if(p_GCParams!=NULL)copy->set_GCParams((XplExpressionlist*)p_GCParams->Clone());
	copy->set_ElementName(this->get_ElementName());
	return (XplNode*)copy;
}
bool XplNewExpression::Write(XplWriter* writer){
	bool result=true;
	//Escribo el encabezado del elemento
	writer->write(DT("<")+this->get_ElementName());
	//Escribo los atributos del elemento
	if(p_GCName != DT("default"))
		writer->write(DT(" GCName=\"")+CODEDOM_Att_ToString(p_GCName)+DT("\""));
	if(p_doc != DT_EMPTY)
		writer->write(DT(" doc=\"")+CODEDOM_Att_ToString(p_doc)+DT("\""));
	if(p_helpURL != DT_EMPTY)
		writer->write(DT(" helpURL=\"")+CODEDOM_Att_ToString(p_helpURL)+DT("\""));
	if(p_ldsrc != DT_EMPTY)
		writer->write(DT(" ldsrc=\"")+CODEDOM_Att_ToString(p_ldsrc)+DT("\""));
	if(p_iny != false)
		writer->write(DT(" iny=\"")+CODEDOM_Att_ToString(p_iny)+DT("\""));
	if(p_inydata != DT_EMPTY)
		writer->write(DT(" inydata=\"")+CODEDOM_Att_ToString(p_inydata)+DT("\""));
	if(p_inyby != DT_EMPTY)
		writer->write(DT(" inyby=\"")+CODEDOM_Att_ToString(p_inyby)+DT("\""));
	if(p_lddata != DT_EMPTY)
		writer->write(DT(" lddata=\"")+CODEDOM_Att_ToString(p_lddata)+DT("\""));
	writer->write(DT(">"));
	//Escribo recursivamente cada elemento hijo
	if(p_type!=NULL)if(!p_type->Write(writer))result=false;
	if(p_init!=NULL)if(!p_init->Write(writer))result=false;
	if(p_GCParams!=NULL)if(!p_GCParams->Write(writer))result=false;
	//Cierro el elemento
	writer->write(DT("</")+this->get_ElementName()+DT(">"));
	return result;
}
XplNode* XplNewExpression::Read(XplReader* reader){
	throw new CodeDOM_NotImplementedException(DT("Read"));
}

//Funciones para obtener Atributos
string XplNewExpression::get_GCName(){
	return p_GCName;
}
string XplNewExpression::get_doc(){
	return p_doc;
}
string XplNewExpression::get_helpURL(){
	return p_helpURL;
}
string XplNewExpression::get_ldsrc(){
	return p_ldsrc;
}
bool XplNewExpression::get_iny(){
	return p_iny;
}
string XplNewExpression::get_inydata(){
	return p_inydata;
}
string XplNewExpression::get_inyby(){
	return p_inyby;
}
string XplNewExpression::get_lddata(){
	return p_lddata;
}

//Funciones para obtener Elementos de Secuencia
XplType* XplNewExpression::get_type(){
	return p_type;
}
XplInitializerList* XplNewExpression::get_init(){
	return p_init;
}
XplExpressionlist* XplNewExpression::get_GCParams(){
	return p_GCParams;
}

//Funciones para setear Atributos
string XplNewExpression::set_GCName(string new_GCName){
	string back_GCName = p_GCName;
	p_GCName = new_GCName;
	return back_GCName;
}
string XplNewExpression::set_doc(string new_doc){
	string back_doc = p_doc;
	p_doc = new_doc;
	return back_doc;
}
string XplNewExpression::set_helpURL(string new_helpURL){
	string back_helpURL = p_helpURL;
	p_helpURL = new_helpURL;
	return back_helpURL;
}
string XplNewExpression::set_ldsrc(string new_ldsrc){
	string back_ldsrc = p_ldsrc;
	p_ldsrc = new_ldsrc;
	return back_ldsrc;
}
bool XplNewExpression::set_iny(bool new_iny){
	bool back_iny = p_iny;
	p_iny = new_iny;
	return back_iny;
}
string XplNewExpression::set_inydata(string new_inydata){
	string back_inydata = p_inydata;
	p_inydata = new_inydata;
	return back_inydata;
}
string XplNewExpression::set_inyby(string new_inyby){
	string back_inyby = p_inyby;
	p_inyby = new_inyby;
	return back_inyby;
}
string XplNewExpression::set_lddata(string new_lddata){
	string back_lddata = p_lddata;
	p_lddata = new_lddata;
	return back_lddata;
}

//Funciones para setear Elementos de Secuencia
XplType* XplNewExpression::set_type(XplType* new_type){
	XplType* back_type = p_type;
	p_type = new_type;
	if(p_type!=NULL){
		p_type->set_ElementName(DT("type"));
		p_type->set_Parent(this);
	}
	return back_type;
}
XplInitializerList* XplNewExpression::set_init(XplInitializerList* new_init){
	XplInitializerList* back_init = p_init;
	p_init = new_init;
	if(p_init!=NULL){
		p_init->set_ElementName(DT("init"));
		p_init->set_Parent(this);
	}
	return back_init;
}
XplExpressionlist* XplNewExpression::set_GCParams(XplExpressionlist* new_GCParams){
	XplExpressionlist* back_GCParams = p_GCParams;
	p_GCParams = new_GCParams;
	if(p_GCParams!=NULL){
		p_GCParams->set_ElementName(DT("GCParams"));
		p_GCParams->set_Parent(this);
	}
	return back_GCParams;
}
XplType* XplNewExpression::new_type(){
	XplType* node = new XplType();
	node->set_ElementName(DT("type"));
	return node;
}
XplInitializerList* XplNewExpression::new_init(){
	XplInitializerList* node = new XplInitializerList();
	node->set_ElementName(DT("init"));
	return node;
}
XplExpressionlist* XplNewExpression::new_GCParams(){
	XplExpressionlist* node = new XplExpressionlist();
	node->set_ElementName(DT("GCParams"));
	return node;
}

}

#endif
